### DISCUSSION TITLE
Good example of showing list of files ?

### DISCUSSION THREAD

**Discussion by XeonG** [unoplatform/uno#17043] [General]
Can't find an example of tree view list of file/folder icons with check box select or something?

datagrids? only seem to find basic stuff

**Comment by jeromelaban** [comment]
Thanks for the question. The playground is a good start: https://playground.platform.uno/#treeview.

You can also try DataGrids with our samples: https://platform.uno/docs/articles/external/uno.samples/doc/samples.html#wct-datagrid and many others.

**Comment by XeonG** [comment]
yeah I mean some better examples... this used to be so much more simpler back with winforms... just drag and drop tons of guides and you'd have a tree view of folder with all the file/folders... now it's some just basic web view with check boxes? seems like a massive downgrade in rich components to be able to build desktop utilities quickly and have a good amount examples and samples to get started with quickly.. 

kinda what I'm missing really, the lack of full training videos to go from start to finished project on a project I'd see enough similarties to what I'd be interested in building myself for something else. ....  it's the lack of that sort of content, I got plenty of other things to do but when that stuff builds up it makes it bit quicker to get started and just get stuck in.. instead it seems like I could waste of lot of time just to find out it's still not really there yet, and at least not for anyone who isn't going to dedicate entire working days to getting something done with it.

It's a right mess out there just trying to find a decent way to show files and folders....and  build something around needing to be able to do that in a desktop app... seems ok if you want to build what is glorified website as a app with buttons and images.. seems all these frameworks are able to that.. but  in tree view lists, icon views and with check box selections etc... amazed winforms even existed sadly I don't care to make just for windows only anymore.

Here are the kind of things I was looking for... 

https://www.telerik.com/blogs/expand-maui-paradise-net-maui-treeview
https://dev.to/icebeam7/creating-a-treeview-control-in-net-maui-49mp
https://www.syncfusion.com/aspnet-core-ui-controls/file-manager


And here something simply done in winforms.. 
https://www.c-sharpcorner.com/article/display-sub-directories-and-files-in-treeview/

even looked better than how things look now with just ugly retarded looking interfaces

**Reply by jeromelaban** [reply]
**@XeonG please [mind your language](https://github.com/unoplatform/uno/blob/master/CODE_OF_CONDUCT.md).**

Thanks for the feedback. We do not have any  sample which covers this use case specifically at this time, but [larger apps exist](https://github.com/files-community/Files), or even [stack overflow](https://stackoverflow.com/questions/76115558/how-to-populate-treeview-in-winui-3). The [TreeView documentation](https://learn.microsoft.com/en-us/windows/apps/design/controls/tree-view) covers how to use it also in details.

The sample I provided above in playground is there to show what a `TreeView` does, and you can have the same running in an app on all platforms, not just web.

If you end up building a component that can serve as a winui/uno building block, please it be known!

**Comment by XeonG** [comment]
> Thanks for the feedback. We do not have any sample which covers this use case specifically at this time

hope you do at some point, more importantly some training that goes over these areas in a start to finish project
